packages/alfa-dom/src/tables/row.ts

```diff
@@ -0,0 +1,289 @@
+    }
+
+    /**
+     * moves xCurrent to the first slot which is not already covered by one of the cells from the row or its context
+     * step 6
+     */
+    private _skipIfCovered(
+      cells: Array<Cell.Building>,
+      yCurrent: number
+    ): Building {
+      if (
+        this._xCurrent < this._width &&
+        cells
+          .concat(this._cells, this._downwardGrowingCells)
+          .some((cell) => cell.isCovering(this._xCurrent, yCurrent))
+      ) {
+        return this._update({ xCurrent: this._xCurrent + 1 })._skipIfCovered(
+          cells,
+          yCurrent
+        );
+      } else {
+        return this;
+      }
+    }
+
+    private _enlargeIfNeeded(): Building {
+      return this._xCurrent === this.width
+        ? this._adjustWidth(this.width + 1)
+        : this;
+    }
+
+    /**
+     * @see/ https://html.spec.whatwg.org/multipage/tables.html#algorithm-for-processing-rows
+     */
+    public static from(
+      tr: Element,
+      cells: Array<Cell.Building> = [],
+      growingCells: Array<Cell.Building> = [],
+      yCurrent: number = 0,
+      w: number = 0
+    ): Result<Building, string> {
+      if (
+        cells.some((cell) =>
+          growingCells.some((growingCell) => cell.equals(growingCell))
+        )
+      )
+        return Err.of("Cells and growing cells must be disjoints");
+      if (tr.name !== "tr") return Err.of("This element is not a table row");
+
+      let children = tr.children().filter(isElementByName("th", "td"));
+
+      // 1
+      // global table height adjusted after building row
+
+      // loop control between 4-5, and 16-17-18
+      return Ok.of(
+        children.reduce(
+          (row, currentCell) =>
+            row
+              // 6 (Cells)
+              ._skipIfCovered(cells, yCurrent)
+              // 7
+              ._enlargeIfNeeded()
+              // 8-14
+              ._addCellFromElement(currentCell, yCurrent)
+              .get(), // can't be an error because children have been filtered
+          // 15 is actually not needed because it will be done as part of step 6 on next loop, and is useless on last element.
+          // 2 is done when creating the row, default value for xCurrent is 0.
+          Building.of(yCurrent, w, 1, tr, [], growingCells)
+            // 3
+            ._growCells(yCurrent)
+        )
+      );
+
+      // return row;
+      // 4 and 16 done after the calls to avoid side effects.
+    }
+
+    public equals(value: unknown): value is this {
+      if (!(value instanceof Building)) return false;
+      const sortedThisCells = this._cells.sort((a, b) => a.compare(b));
+      const sortedValueCells = value._cells.sort((a, b) => a.compare(b));
+      const sortedThisDGCells = this._downwardGrowingCells.sort((a, b) =>
+        a.compare(b)
+      );
+      const sortedValueDGCells = value._downwardGrowingCells.sort((a, b) =>
+        a.compare(b)
+      );
```

Comment:
`Array#sort()` sorts in place so there's a) no need to use the return value and b) no need to do it more than once during construction.